demo-circuit/src/naive_threshold_sig_w_key_rotation/tests.rs

```diff
@@ -0,0 +1,897 @@
+use algebra::{PrimeField, UniformRand, Field, ToConstraintField, ProjectiveCurve};
+use cctp_primitives::{utils::commitment_tree::DataAccumulator, 
+    proving_system::{init::{get_g1_committer_key, load_g1_committer_key}, compute_proof_vk_size, ProvingSystem}};
+use primitives::{FieldBasedHash, FieldBasedSignatureScheme, FieldBasedMerkleTree, schnorr::field_based_schnorr::FieldBasedSchnorrSignature};
+use r1cs_core::debug_circuit;
+use rand::{rngs::OsRng, thread_rng};
+use serial_test::serial;
+
+use crate::{type_mapping::*, NaiveThresholdSignatureWKeyRotation, NULL_CONST};
+use rand::Rng;
+
+use crate::{MAX_SEGMENT_SIZE, SUPPORTED_SEGMENT_SIZE};
+pub const VALIDATORS_SIZE: usize = 6;
+
+#[derive(Copy, Clone, Debug)]
+enum TestDataType {
+    AllRandomFirstCert,
```

Comment:
Let's add test cases also for wrong updates on signing/master keys

Reply:
test_naive_threshold_sig_w_key_rotation_circuit_wrong_signature_after_updated_keys

Reply:
Fixed with new tests.